src/scripts/contract_allocations.py

```diff
@@ -0,0 +1,55 @@
+from src.constants import NODE_URL
+from src.fetch.contracts import EvmAccountInfo
+from src.files import AllocationFiles, NetworkFile, File
+from src.generate.merkle_data import MerkleLeaf
+
+
+def non_wallet_allocation(network: str, allocation_file: File) -> int:
+    allocations = MerkleLeaf.load_from(allocation_file)
+    account_info = EvmAccountInfo(
+        node_url=NODE_URL[network],
+        addresses=[a.Account for a in allocations],
+        network=network
+    )
+    contracts = account_info.contracts(load_from=NetworkFile("contracts.txt"))
+    not_wallets = account_info.get_non_wallets(
+        [c for c, is_contract in contracts.items() if is_contract]
+    )
+    print(
+        f"found {len(not_wallets)} contract allocations to non-wallets on {network}.\n"
+        f"printing Top 3"
+    )
+    sorted_contract_allocations = sorted([
+        a for a in allocations
+        if contracts[a.Account] is True
+    ], key=lambda t: t.Airdrop, reverse=True)
+
+    airdrop_total, found = 0, 0
+    for alloc in sorted_contract_allocations:
+        if alloc.Account in not_wallets:
+            # proof that this only affects GNO allocations!
+            gno_allocation = alloc.Airdrop + alloc.GnoOption
+            assert alloc.total() - gno_allocation == 0
+            airdrop_total += alloc.Airdrop
+            if found < 3:
+                found += 1
+                print(
+                    f"{found}. {alloc.Account} - {(alloc.Airdrop / 1e24):.3f}M"
+                )
```

Comment:
This seems like a magic number to me. I'm guessing there are 3 "special" addresses which we are interested here.
To make it to read for people without background knowledge something like this could make sense:
```python
special_addresses = set(
    "0x...1", # reason why this is special
    # ...
)
for alloc in sorted_contract_allocations:
    if alloc.Account in not_wallets:
        if alloc.Account in special_addresses:
            print(
                f"{found}. {alloc.Account} - {(alloc.Airdrop / 1e24):.3f}M"
            )
        # ... rest of the loop

```

Reply:
They aren't really special, they are just the top 3. The print statement above says printing top three.

Reply:
Oh, I missed that. In that case ignore what I said. 🙈